### ISSUE TITLE
naming conventions / folder structure

### ISSUE THREAD

**Issue by fxck** [meenie/angular2-store-example#7]
## use case
list of users, detail of a user

## structure 
- routes
 - UsersListRoute.ts
 - UsersDetailRoute.ts
- containers
 - UsersListContainer.ts
 - UsersDetailContainer.ts
- components
 - UsersList.ts
 - UsersListItem.ts
 - UsersDetail.ts

### routes
- serves just like a wrapper to get the route working
- may include containers and graphical elements that don't have any state / actions

### containers
- smart components, you inject your services here(standard smart component)

### components
- dumb components, only render data passed to them by attributes(standard smart component)

better ideas? thoughts?

**Comment by meenie** [comment]
I like where this is going, but I'm wondering if we can somehow work in sub-containers/components.  Meaning, you have a large app, and there are a couple of standalone containers that fit underneathe top level containers.  For example, your app runs an online store.  In the backend, you'll have top level "Products" area where you list all of your products to be managed.  Underneath that, you need an "Edit Product" screen.  "ProductsList" and "ProductsDetail" are a parent/child  relationship and will reflect that in the routes.  So wouldn't you want to nest the "ProductsDetail" underneath "ProductsList"?

**Comment by fxck** [comment]
I don't think I would, as long as the detail is on a different URL, I don't really think of it as a child of the list. What would make sense to me though, is if detail and list had same parent/containers, that is, say, just `UsersContainer`.

**Comment by meenie** [comment]
The urls I was thinking were:
 - `/users` <-- `UsersList` container
 - `/users/1` <-- `UsersDetail` container

So they naturally sit underneath each other.  Or were you thinking something different?

**Comment by fxck** [comment]
Are they though? In this situation I would have 

```
@RouteConfig([{
    path: '/users/...',
    component: UsersComponent,
    as: 'Users',
}])
class AppComponent {}


@RouteConfig([
	{ 
		path: '/list', 
		component: UsersListComponent,
		as: 'UsersList'	
	},
	{ 
		path: '/detail/:id', 
		component: UsersDetailComponent,
		as: 'UsersDetail'		
	}	
])
class UsersComponent {}
```

**Comment by meenie** [comment]
No, not necessarily, and I do like that structure.  Has parent/child but also keeps it a bit flat.  What would be the folder structure look like?  Also, shouldn't the classes be `UsersContainer`, `UsersListContainer`, `UsersDetailContainer`?

**Comment by fxck** [comment]
Yea I was just copying some of my old code. :)

**Comment by meenie** [comment]
>Yea I was just copying some of my old code. :)

:+1:

**Comment by robwormald** [comment]
i like @fxck's structure idea here - if you've used UI-router in ng1, its akin to an "abstract" state that encompasses a set of features/functionality.

**Comment by Merott** [comment]
I'm setting up `ngrx` in a new project, and I'm trying to set up the directory structure and define the conventions. This is how I've done the directory structure so far.

```
app/
├── components
├── containers
├── resources
│   ├── articles.resource.ts
│   └── users.resource.ts
└── store
    ├── data.articles
    │   ├── data.articles.model.spec.ts
    │   ├── data.articles.model.ts
    │   ├── data.articles.reducer.spec.ts
    │   └── data.articles.reducer.ts
    ├── data.users
    │   ├── data.users.model.spec.ts
    │   ├── data.users.model.ts
    │   ├── data.users.reducer.spec.ts
    │   └── data.users.reducer.ts
    ├── ui.favorites
    │   ├── ui.favorites.model.spec.ts
    │   ├── ui.favorites.model.ts
    │   ├── ui.favorites.reducer.spec.ts
    │   └── ui.favorites.reducer.ts
    ├── ui.home
    │   ├── ui.home.model.spec.ts
    │   ├── ui.home.model.ts
    │   ├── ui.home.reducer.spec.ts
    │   └── ui.home.reducer.ts
    ├── store.spec.ts
    └── store.ts
```

I'm putting `model` and `reducer` files together under the same directory, all within a root `store` directory.  The `store.ts` module will create the store and export `storeProviders`.

Under `resources`, I'm creating Resource services, where actual API calls happen. The models use the resources to fetch data.

The motivation behind this structure is to keep everything related to the store together in one place, and to make clear distinction between ui and data states.

I'm new to this, and that's my plan. Does anything seem off to you? Also, I have a few other questions:

1. Should I be storing the `ui` state the way I've done here?
2. Where to put component state, e.g. the `disabled` state of a button?
